### DISCUSSION TITLE
Dependency Inception

### DISCUSSION THREAD

**Discussion by kwelch** [renovatebot/renovate#28993] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Bitbucket

### Please tell us more about your question or problem

We have a custom image wrapper around base node and java images. The image name include the major version of the language it is wrapping (ie. `golden-image-node-20`, `golden-image-node-18`,  `golden-image-java-17`, `golden-image-java-21`). These images then also follow their own version tagging. 

Out of the box as expected Renovate is able to manage the image tag version with no issues. 

Additionally, I was able to make a custom manager that could also manage the version within the image name. 

However, I am not seeing a path forward to link these into a single PR such that the image version is properly adusted based on a `node` or `java` version change. 

```diff
-FROM golden-image-node-18:2.0.13
+FROM golden-image-node-20:2.0.13 
```

Unfortunately, the above PR from Renovate will not work, as there is no version `2.0.13` for `golden-image-node-20`. 

Here is the regex manager:
```json
{
      "description": "Custom manager for node micros golden images to upgrade node versions",
      "customType": "regex",
      "fileMatch": [
        "(^|/|\\.)([Dd]ocker|[Cc]ontainer)file$",
        "(^|/)([Dd]ocker|[Cc]ontainer)file[^/]*$"
      ],
      "datasourceTemplate": "node-version",
      "matchStrings": [
        "FROM golden-image-(?<depName>node)\\-(?<currentValue>[\\d\\.]+):"
      ]
    }
```

Out self-hosted instance is running version `37.240.1`

Any help or ideas would be appreciated. 

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
Adding a _replacement_ for node18 to node20 might work better. Then you can specify the replacementVersion too

**Reply by kwelch** [reply]
Oh nice! They are actively being developed but that would help get over the hurdle of the first upgrade across the company.

**Comment by HonkingGoose** [comment]
> Our self-hosted instance is running version `37.240.1`

By the way, you should update your version of Renovate. Your version was released on March 12th, that's nearly two months ago. It probably won't fix your problem, but at least you're using current stuff then. :smile:

**Reply by kwelch** [reply]
Thanks! we have renovate renovating that via regex manager, but we only upgrade monthly. We just missed the April upgrade due to an offsite. We have patches for metrics so taking on upgrades is not always super simple.
